### PR TITLE
fix: Workaround for Mizrahi bank API bug which returns transactions before the start date

### PR DESCRIPTION
**Pull request by alonl**
For some reason Mizrahi bank API returns transactions before the requested "from" date (start date). From what I've noticed, usually it is one additional transaction, and it doesn't matter what its date is.

With YNAB exporter in [Caspion](https://github.com/brafdlog/caspion) it does a real mess since it makes those transactions being imported again and again and creates duplicates.

This fix is a workaround for this bug of Mizrahi Bank API which double checks the transaction date is within range.

### CODE REVIEW

File: src/scrapers/mizrahi.ts [hunk 1]
```diff
@@ -172,10 +174,14 @@ class MizrahiScraper extends BaseScraperWithBrowser {
     const relevantRows = response.body.table.rows.filter((row) => row.RecTypeSpecified);
     const oshTxn = convertTransactions(relevantRows);
 
+    // workaround for a bug which the bank's API returns transactions before the requested start date
+    const startMoment = getStartMoment(this.options.startDate);
+    const oshTxnAfterStartDate = oshTxn.filter(txn => !moment(txn.date).isBefore(startMoment));
```
Comment (eshaham) [inline src/scrapers/mizrahi.ts:179]:
```suggestion
    const oshTxnAfterStartDate = oshTxn.filter(txn => moment(txn.date). isSameOrAfter(startMoment));
```

**Review by eshaham — CHANGES_REQUESTED** [review]
Thanks for contributing 😄 
Question about the bug with the bank's API - are the transactions returned missed by more than one day?
Meaning, if I specify the start date as 10/7, will it also return transactions from 5/7?

**Comment by alonl** [comment]
> Thanks for contributing 
> Question about the bug with the bank's API - are the transactions returned missed by more than one day?
> Meaning, if I specify the start date as 10/7, will it also return transactions from 5/7?

Thank you! 
Yes. From my observation it seems that it returns one additional transaction, which is the last one that's happened before the start date. It can be one day before or even 2 weeks.

**Comment by eshaham** [comment]
@alonl the lint job is failing - can you run `npm run lint` locally and fix the lint errors?

**Comment by alonl** [comment]
> @alonl the lint job is failing - can you run `npm run lint` locally and fix the lint errors?

Fixed, sorry.

**Review by eshaham — APPROVED** [review]
looking good, thanks 👌

**Pull request merged**